src/utils.jl

```diff
@@ -9,6 +9,63 @@ macro check_args(D, cond)
     end
 end
 
+isuncountable(x) = missing
+isuncountable(x::Number) = typeof(float(one(x))) ≡ typeof(one(x))
+
+macro check_uncountable(D, var)
+    quote
+        if !isuncountable($(esc(var)))
+            throw(ArgumentError(string($(string(D)),
+                                       ": the type of variable ",
+                                       $(string(var)), " (",
+                                       string(typeof($(esc(var)))),
+                                       ") is not uncountable.")))
+        end
+    end
+end
+
+macro check_countable(D, var)
+    quote
+        if isuncountable($(esc(var)))
+            throw(ArgumentError(string($(string(D)),
+                                       ": the type of variable ",
+                                       $(string(var)), " (",
+                                       string(typeof($(esc(var)))),
+                                       ") is not countable.")))
+        end
+    end
+end
+
+
+## a type to indicate zero vector
```

Comment:
@richardreeve I messed up my conflict merge, could you remove things related to ZeroVector in this file, it was deleted in:
https://github.com/JuliaStats/Distributions.jl/pull/1020